pallets/subscriptions/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+//! # Subscription Module
+//!
+//! The Subscription module allows supporters of any space that represents a creator or
+//! a nonprofit organization to contribute financially (with native tokens) a daily, weekly,
+//! monthly, quarterly or yearly basis. This module allows content creators to monetize
+//! their content via recurring payments from their supporters.
+//!
+//! This pallet provides a way for creators to create a list of subscription plans (aka levels, tiers)
+//! and specify a different price and time period per each plan. There are several pre-built
+//! subscription periods: `Daily`, `Weekly`, `Monthly`, `Quarterly` and `Yearly`.
+//!
+//! This pallet uses Substrate's Schedule pallet to schedule recurring transfers from supporters'
+//! (patrons') wallets to creators' wallets.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Encode, Decode};
+use sp_std::prelude::*;
+use sp_runtime::RuntimeDebug;
+
+use frame_support::{
+	decl_module, decl_storage, decl_event, decl_error, ensure,
+	dispatch::{Dispatchable, DispatchResult},
+	traits::{
+		Get, Currency, ExistenceRequirement,
+		schedule::Named as ScheduleNamed, LockIdentifier,
+	}
+};
+use frame_system::{self as system, ensure_signed, ensure_root};
+
+use pallet_spaces::Module as Spaces;
+use pallet_utils::{Module as Utils, SpaceId, Content, WhoAndWhen, vec_remove_on};
+
+/*#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;*/
+
+pub mod functions;
+
+const SUBSCRIPTIONS_ID: LockIdentifier = *b"subscrip";
+
+pub type SubscriptionPlanId = u64;
+pub type SubscriptionId = u64;
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub enum SubscriptionPeriod<BlockNumber> {
+	Daily,
+	Weekly,
+	Monthly,
+	Quarterly,
+	Yearly,
+	Custom(BlockNumber), // Currently not supported
```

Comment:
I think we need to put `Custom(BlockNumber)` as the first option?

Reply:
AFAIR, we decided to move it to the end of the enum, because this option is quite unstable and shouldn't be the first one available. I would say, this may be deleted with time for security purposes, so I suppose we should leave it at the end.